src/quotamodel.c

```diff
@@ -1503,6 +1504,30 @@ do_load_quotas(void)
 			targetOid = primaryOid;
 		}
 
+		int cacheid;
+		switch (quotaType)
+		{
+			case NAMESPACE_QUOTA:
+			case NAMESPACE_TABLESPACE_QUOTA:
+				cacheid = NAMESPACEOID;
+				break;
+			case ROLE_QUOTA:
+			case ROLE_TABLESPACE_QUOTA:
+				cacheid = AUTHOID;
+				break;
+			case TABLESPACE_QUOTA:
+				cacheid = TABLESPACEOID;
+				break;
+			default:
+				Assert(false); /* never reach here */
+		}
+
+		if (!SearchSysCacheExists1(cacheid, ObjectIdGetDatum(targetOid)))
+		{
+			cleanConfigTables = true;
+			continue;
```

Comment:
Is this line really necessary?

Reply:
Shouldn't we update quotas in this case?

Reply:
there is no object for which a quota is set. So it is pointless to update quotas in this case